sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -719,7 +719,7 @@ object StringTrim {
 
     _FUNC_(TRAILING FROM str) - Removes the trailing space characters from `str`.
 
-    _FUNC_(str, trimStr) - Remove the leading and trailing `trimStr` characters from `str`.
+    _FUNC_(trimStr, str) - Remove the leading and trailing `trimStr` characters from `str`.
```

Comment:
I'm thinking about if we should remove this line (as well as the example) and hide this API. This was definitely a mistake and it's not even consistent with the scala version in `functions.scala`. For backward compatibility, we should keep this mistake but don't expose it too publicly.

I think `trimStr FROM str` should be the recommended API at the SQL side.

Reply:
cc @wangyum @viirya @maropu

Reply:
I personally like the explicit format like `trimStr FROM str` and the idea to hide the api in future releases look reasonable to me. If we fix it that way, I think its better to fix the other string functions consistently, e.g., `substring` and `overlay`.

These string functions are pretty common in dbms-like systems, so I feel throwing an exception is better than returning a wrong answer when users just copy&paste a query including these string functions from another system.

Reply:
Hmm, this looks like a mistake at all. Seems no other feasible solution for this. I'm fine to remove it from doc. One question is shall we add some words in migration guide? Can we say we deprecate it and then we will make correct one in future version?